test/postgresql.test.js

```diff
@@ -820,6 +820,53 @@ describe('Serial properties', function() {
   });
 });
 
+describe('Updating of array properties', function() {
+  let db;
+
+  before(function() {
+    db = global.getSchema();
+  });
+
+  it('should return a valid array', function(done) {
+    const schema =
+      {
+        'name': 'TestWithArray',
+        'properties': {
+          'id': {
+            'type': 'number', 'id': true, 'generated': true,
+          },
+          'productCodes': {
+            'type': ['string'],
+          },
+        },
+      };
+    const models = db.modelBuilder.buildModels(schema);
+    const Model = models['TestWithArray'];
+    const count = 0;
+    Model.attachTo(db);
+
+    db.automigrate(function(err, data) {
+      async.series([
+        function(callback) {
+          Model.destroyAll(callback);
+        },
+        function(callback) {
+          Model.create({productCodes: ['AA', 'AB']}, callback);
```

Comment:
We support the promise chain of CRUD operations, example see 
https://github.com/strongloop/loopback-datasource-juggler/blob/master/test/crud-with-options.test.js#L149

So you can write the chain calls as 

```js
Post.create(params)
  .then(()=> {
     return Post.updateAll(params)
   })
  .then(()=> {
      // other function calls
   })
```